lib/THNN/generic/THNN.h

```diff
@@ -91,4 +104,50 @@ TH_API void THNN_(L1Cost_updateGradInput)(
           THTensor *gradOutput,
           THTensor *gradInput);
 
+TH_API void THNN_(LeakyReLU_updateOutput)(
+          THNNState *state,
+          THTensor *input,
+          THTensor *output,
+          real negval,
+          bool inplace);
+TH_API void THNN_(LeakyReLU_updateGradInput)(
+          THNNState *state,
+          THTensor *input,
+          THTensor *gradOutput,
+          THTensor *gradInput,
+          real negval,
+          bool inplace);
+
+TH_API void THNN_(LogSigmoid_updateOutput)(
+          THNNState *state,
+          THTensor *input,
+          THTensor *output,
+          THTensor *buffer);
+TH_API void THNN_(LogSigmoid_updateGradInput)(
+          THNNState *state,
+          THTensor *input,
+          THTensor *gradOutput,
+          THTensor *gradInput,
+          THTensor *buffer);
+
+TH_API void THNN_(LogSoftMax_updateOutput)(
+          THNNState *state,
+          THTensor *input,
+          THTensor *output);
+TH_API void THNN_(LogSoftMax_updateGradInput)(
+          THNNState *state,
+          THTensor *input,
+          THTensor *gradOutput,
+          THTensor *gradInput,
+          THTensor *output);
+
+TH_API void THNN_(LookupTable_accGradParameters)(
+          THNNState *state,
+          THLongTensor *input,
+          THTensor *gradOutput,
+          THTensor *gradWeight,
+          real lr,
+          bool shouldScaleGradByFreq,
+          THLongTensor* count);
```

Comment:
Wouldn't it be better to use `THIndexTensor` instead ? You have changed everywhere from `int` to `long`, I thought you would want to make it consistent with `THIndexTensor`